glib/src/match_info.rs

```diff
@@ -1,9 +1,380 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
-use crate::{translate::*, IntoGStr, MatchInfo};
+use crate::{translate::*, GStr, IntoGStr, Regex};
 use std::{mem, ptr};
 
-impl MatchInfo {
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[repr(transparent)]
+pub struct MatchInfo<'input> {
+    inner: crate::shared::Shared<ffi::GMatchInfo, Self>,
+    _phantom: std::marker::PhantomData<&'input GStr>,
+}
+
+impl<'input> MatchInfo<'input> {
+    #[doc = "Return the inner pointer to the underlying C value."]
+    #[inline]
+    pub fn as_ptr(&self) -> *mut ffi::GMatchInfo {
+        unsafe { *(self as *const Self as *const *const ffi::GMatchInfo) as *mut ffi::GMatchInfo }
+    }
+    #[doc = "Borrows the underlying C value."]
+    #[inline]
+    pub unsafe fn from_glib_ptr_borrow<'a>(ptr: *const *const ffi::GMatchInfo) -> &'a Self {
+        &*(ptr as *const Self)
+    }
+}
+impl<'input> std::clone::Clone for MatchInfo<'input> {
```

Comment:
This can be derived (as could `Copy`)

Reply:
You're right, it won't be possible to implement `Copy` because of the `Drop` impl. But as this is basically a glorified reference now, you don't really need the reference counting on `Clone`, it's just needed because of the `Drop` impl :)